controllers/trainingjob/trainingjob_controller.go

```diff
@@ -170,6 +176,25 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		}
 	}
 
+	// Check  sagemaker describe output for each debug jobs and
+	// generates the update if k8s status for corresponding debug jobs differ.
+	// Initially status will be empty so we will generate update for all debug jobs
+	if len(ctx.TrainingJobDescription.DebugRuleEvaluationStatuses) != len(ctx.TrainingJob.Status.DebugRuleEvaluationStatuses) {
+		for _, _ = range ctx.TrainingJobDescription.DebugRuleEvaluationStatuses {
+			if err = r.addDebugRuleEvaluationStatusesToStatus(ctx); err != nil {
+				return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add debug job statuses to status"))
+			}
+		}
+	} else {
+		for i, debugJob := range ctx.TrainingJobDescription.DebugRuleEvaluationStatuses {
+			if string(debugJob.RuleEvaluationStatus) != controllers.GetOrDefault(ctx.TrainingJob.Status.DebugRuleEvaluationStatuses[i].RuleEvaluationStatus, "") {
+				if err = r.addDebugRuleEvaluationStatusesToStatus(ctx); err != nil {
```

Comment:
Same as above, if `addDebugRuleEvaluationStatusesToStatus` adds all the rule evaluation statuses to the status then we only need to run it once (break out of the loop).

Reply:
In that case I'll just un-conditionally call that method.

Reply:
Bumping this comment. Seems like we still have unnecessary looping.

Reply:
I think you missed the `break` in line 192. the loop is necessary to find the job where k8s status and sm status are different.